src/zenml/zen_stores/sql_zen_store.py

```diff
@@ -1005,6 +1042,111 @@ def _initialize(self) -> None:
                 self._backup_secrets_store.config
             )
 
+    def restore_database(self) -> None:
+        """Restore the database from a backup."""
+        # if a backup file does not exist, return
+        assert self.db_backup_file_path is not None
+        db_name = self.db_backup_file_path.split("/")[-1].split(".")[0]
+        if not os.path.exists(self.db_backup_file_path):
+            logger.debug("No backup found. No restore performed.")
+            return
+
+        with self.engine.begin() as connection:
+            # drop the database if it exists
+            connection.execute(text(f"DROP DATABASE IF EXISTS {db_name}"))
+
+            # then create the database
+            connection.execute(text(f"CREATE DATABASE {db_name}"))
+
+            connection.execute(text(f"USE {db_name}"))
+
+        with self.engine.begin() as connection:
+            # read the dump file and execute it
+            with open(self.db_backup_file_path, "r") as f:
+                contents = f.read()
+
+            # remove any escaped characters like \n and \t and split based on ; but keep the ; at the end of each sentence
+            contents = (
+                contents.replace("\\n", "").replace("\\t", "").split(";")
+            )
+
+            # execute each statement
+            for statement in contents:
+                if statement.strip() != "":
+                    connection.execute(text(statement + ";"))
+
+        logger.debug(f"Database restored from {self.db_backup_file_path}")
+
+        # delete the database dump file
+        os.remove(self.db_backup_file_path)
+
+    def backup_database(self) -> None:
+        """Backup the database to a file."""
+        db_name = self.db_backup_file_path.split("/")[-1].split(".")[0]
+
+        # create the backup directory if it does not exist
+        if not os.path.exists(self.config.backup_directory):
+            os.makedirs(self.config.backup_directory)
+
+        if self.config.driver == SQLDatabaseDriver.SQLITE:
+            # dump the database to a file
+            os.system(
+                f"sqlite3 {self.config.database} .dump > {self.db_backup_file_path}"
+            )
```

Comment:
```suggestion
            )
            return
```
otherwise you fall over to the mysql code.